src/java.base/share/classes/java/util/Base64.java

```diff
@@ -741,6 +741,67 @@ private int decodedOutLength(byte[] src, int sp, int sl) {
             return 3 * (int) ((len + 3L) / 4) - paddings;
         }
 
+        /**
+         * Decodes base64 characters, and returns the number of data bytes
+         * written into the destination array.
+         *
+         * It is the fast path for full 4-byte to 3-byte decoding w/o errors.
+         *
+         * decodeBlock() can be overridden by an arch-specific intrinsic.
+         * decodeBlock can choose to decode all, none, or a variable-sized
+         * prefix of the src bytes.  This allows the intrinsic to decode in
+         * chunks of the src that are of a favorable size for the specific
+         * processor it's running on.
+         *
+         * If the intrinsic function does not process all of the bytes in
+         * src, it must process a multiple of four of them, making the
+         * returned destination length a multiple of three.
+         *
+         * If any illegal base64 bytes are encountered in src by the
+         * intrinsic, the intrinsic must return the actual number of valid
+         * data bytes already written to dst.  Note that the '=' pad
+         * character is treated as an illegal Base64 character by
+         * decodeBlock, so it will not process a block of 4 bytes
+         * containing pad characters.
+         *
+         * Given the parameters, no length check is possible on dst, so dst
+         * is assumed to be large enough to store the decoded bytes.
+         *
+         * @param  src
+         *         the source byte array of Base64 encoded bytes
+         * @param  sp
+         *         the offset into src array to begin reading
+         * @param  sl
+         *         the total length of source array
+         * @param  dst
```

Comment:
Please update the comment for `sl`.  sl is the offset (exclusive) past the last byte to be converted.

Reply:
Ok, will change.